### ISSUE TITLE
`:GoDef` on a variable in Vim 8.0 with Vim-Go 1.10 does not handle errors gracefully

### ISSUE THREAD

**Issue by leighmcculloch** [fatih/vim-go#1130]
I think `:GoDef` on a variable in Vim 8.0 with Vim-Go 1.10 is not handling errors gracefully.

```
vim-go: FAILED
Error detected while processing function 279[35]..280:
line 20:
E716: Key not present in Dictionary: bang
E15: Invalid expression: !empty(errors) && !self.bang
```

The two locations I could find in the vim-go code that have `!empty(errors) && !self.bang` are handling errors, and I think `bang` does not exist on `self`. I'm not sure what the underlying error is, but I can investigate further if I am given some guidance.

### Actual behavior

`:GoDef` on a variable, and after a few second the following error will display and will show for a couple seconds at most:

![screen shot 2016-11-24 at 3 00 01 pm](https://cloud.githubusercontent.com/assets/351529/20611375/e6da1586-b25d-11e6-8d27-6ae6a7543fdd.png)

After the error disappears a location list window will display which I can then select in to go to the definition. But it doesn't navigate to the definition automatically.

### Expected behavior

Prior to upgrading to the latest version of VIM I would use GoDef to go directly to the definition. I'd be taken there directly and wouldn't have to select the entry in the location list, and there would be no error.

### Steps to reproduce:

Please create a reproducible case of your problem. Re produce it 
with a minimal `vimrc` with all plugins disabled and only `vim-go`
enabled:

1. Setup a new instance running Ubuntu 16.10 (I use these scripts [@leighmcculloch/devenv](https://github.com/leighmcculloch/devenv))
2. Use the [@leighmcculloch/vim_dotfiles](https://github.com/leighmcculloch/vim_dotfiles)'s vimrc
3. Open any Go file.
4. `:GoDef` on a variable.

### Configuration

* vimrc: https://github.com/leighmcculloch/vim_dotfiles/blob/b51e84caca12191a777d62fde3faf55a26b4bfc0/vimrc
* vim version:
```
$ vim --version
VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Nov 24 2016 21:46:31)
Included patches: 1-101
Compiled by root@devenv
Huge version without GUI.  Features included (+) or not (-):
+acl             +file_in_path    +mouse_sgr       +tag_old_static
+arabic          +find_in_path    -mouse_sysmouse  -tag_any_white
+autocmd         +float           +mouse_urxvt     -tcl
-balloon_eval    +folding         +mouse_xterm     +termguicolors
-browse          -footer          +multi_byte      +terminfo
++builtin_terms  +fork()          +multi_lang      +termresponse
+byte_offset     -gettext         -mzscheme        +textobjects
+channel         -hangul_input    +netbeans_intg   +timers
+cindent         +iconv           +num64           +title
-clientserver    +insert_expand   +packages        -toolbar
-clipboard       +job             +path_extra      +user_commands
+cmdline_compl   +jumplist        -perl            +vertsplit
+cmdline_hist    +keymap          +persistent_undo +virtualedit
+cmdline_info    +lambda          +postscript      +visual
+comments        +langmap         +printer         +visualextra
+conceal         +libcall         +profile         +viminfo
+cryptv          +linebreak       -python          +vreplace
+cscope          +lispindent      -python3         +wildignore
+cursorbind      +listcmds        +quickfix        +wildmenu
+cursorshape     +localmap        +reltime         +windows
+dialog_con      -lua             +rightleft       +writebackup
+diff            +menu            -ruby            -X11
+digraphs        +mksession       +scrollbind      -xfontset
-dnd             +modify_fname    +signs           -xim
-ebcdic          +mouse           +smartindent     -xpm
+emacs_tags      -mouseshape      +startuptime     -xsmp
+eval            +mouse_dec       +statusline      -xterm_clipboard
+ex_extra        -mouse_gpm       -sun_workshop    -xterm_save
+extra_search    -mouse_jsbterm   +syntax          
+farsi           +mouse_netterm   +tag_binary      
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
       defaults file: "$VIMRUNTIME/defaults.vim"
  fall-back for $VIM: "/usr/local/share/vim"
Compilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H     -g -O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1       
Linking: gcc   -L/usr/local/lib -Wl,--as-needed -o vim        -lm -ltinfo -lnsl  -ldl           
```
* vim-go version:
```
1.10
```
* go version:
```
$ go version
go version go1.7.3 linux/amd64
```

**Comment by fatih** [comment]
Hi @leighmcculloch 

Unfortunately I can't reproduce your error. Calling `:GoDef` works fine. Also there is no `bang` feature in `:GoDef`. The error you receive comes from somewhere else, not `:GoDef`. I assume you're calling `:GoBuild` maybe? or something similiar?  However I've tried to fix it with this commit anyway https://github.com/fatih/vim-go/commit/398680c1f73d30763093dc206f133e69f155829d

**Comment by enodata** [comment]
Hi @fatih 

I had the same issue in vim 8.0.134 on FreeBSD 11.0-STABLE and https://github.com/fatih/vim-go/commit/398680c1f73d30763093dc206f133e69f155829d fixes it for me, thanks. One small issue though - :GoDef still opens quickfix windows even if there's only one function definition, is there a way to skip opening quickfix if function to go to is unambigious?

**Comment by leighmcculloch** [comment]
@fatih: Thanks. I've just used the latest vim-go with latest vim from src and the cursor now goes to the right place! I'm still seeing a vim-go FAILED message, but I'm not seeing any actual behavior that's not correct now. I'm definitely only running `:GoDef`, not `:GoBuild`.

I just slimmed down my vimrc plugins in batches to see if the vim-go: FAILED message was being contributed by another plugin. Turns out if I remove the `fatih/molokai` plugin, it goes away. Is it possible your other molokai plugin is not compatible with vim-go?
